swarm/api/http/server.go

```diff
@@ -201,6 +201,7 @@ func (s *Server) HandleBzzGet(w http.ResponseWriter, r *http.Request) {
 		defer reader.Close()
 
 		w.Header().Set("Content-Type", "application/x-tar")
+		w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s.tar\"", path.Base(r.URL.Path)))
```

Comment:
two issues with this:
1. do not use r.URL as it will contain the entire requested URI e.g. `/bzz:/somename.eth/something` etc. please use the `uri` variable above which comes from the middleware that parses the bzz uri. it also has a `Path` member which you can use to get to the path inside the bzz uri.
2. there could be a situation where you would be downloading a tarball but the filename might be empty. in that case, you should fallback to the `Addr` from `uri` (or `untitled`)

Reply:
even if its one file - if there's an enclosing manifest, i think there should be a filename there too (need to double check. in any case - there should be - let's assume that)